### ISSUE TITLE
[BUG] Unable to complete setup for CRC

### ISSUE THREAD

**Issue by AlphaMalik** [code-ready/crc#2446]
### General information

  * OS: Windows
  * Hypervisor: Hyper-V 
  * Did you run `crc setup` before starting it (Yes/No)? Yes
  * Running CRC on: Laptop

## CRC version
```> crc version
CodeReady Containers version: 1.28.0+08de64bd
OpenShift version: 4.7.13 (embedded in executable)
```
  
## CRC status
```crc status
Machine does not exist. Use 'crc start' to create it
```

## CRC config
```- consent-telemetry                     : yes
```

Steps to reproduce the issue:
1. Execute crc setup on Windows command prompt
2. this is the output:

> INFO Checking if crc-admin-helper executable is cached
> INFO Checking for obsolete admin-helper executable
> INFO Checking if running on a supported CPU architecture
> INFO Checking minimum RAM requirements
> INFO Checking if running in a shell with administrator rights
> INFO Checking Windows 10 release
> INFO Checking Windows edition
> INFO Checking if Hyper-V is installed and operational
> INFO Checking if crc-users group exists
> INFO Checking if current user is in Hyper-V group and crc-users group
> INFO Adding current user to group
> INFO Will run as admin: adding current user to crc-users group

Please reboot your system and run 'crc setup' to complete the setup process
3. Instructions say reboot the system and run crc setup again.
4. I have restarted my system like a bajillion times but it still ends up in the same location as step 2.

Here's the CRC log:
Link to gist: https://gist.github.com/AlphaMalik/febdfe5dfb6aea5125b13122a601e1cd

**Comment by gbraad** [comment]
> INFO Will run as admin: adding current user to crc-users group

Can you verify if the current user is a member of the Hyper-V admin group and crc-users?

**Comment by guillaumerose** [comment]
I suspect the username we found in crc is not the good one. We use the env var. USERNAME. 
Is '0B5293649' your username?

Perhaps you can run crc setup with a forced USERNAME env variable.

**Comment by DerekBennett** [comment]
Thanks for the tip, @guillaumerose! I had exactly the same problem. To diagnose it, I ran:
`crc setup --log-level debug`
This showed me that the failed command was like this:
`Get-LocalGroupMember -Name crc-users -Member 'user.name'`
Executing without the Member parameter showed something like this:
`ObjectClass Name                 PrincipalSource`
`\----------- ----                 ---------------`
`User        MY_CORPORATE_DOMAIN\user.name ActiveDirectory`

Note that the searched-for USERNAME does not match the actual domain-qualified USERNAME.

To get crc to run all the way through, I re-set my USERNAME environment variable locally with the following PowerShell snippet:
`$env:USERNAME="${env:USERDOMAIN}\$env:{USERNAME}"`

**Comment by AlphaMalik** [comment]
Thanks @DerekBennett ... that fixed it... found out the env:USERNAME was not including the USERDOMAIN. And the user added to the crc group had the USERDOMAIN appended to the USERNAME. Once I executed the snippet you provided above I was able to get past that line in the setup.

Used this snippet to setup the USERNAME environment variable that finally got it to work: 
`$env:USERNAME="${env:USERDOMAIN}\${env:USERNAME}"`

This issue can now be closed.

**Comment by AlphaMalik** [comment]
thanks @guillaumerose ... yep, yes... that was the issue.

**Comment by gbraad** [comment]
> User MY_CORPORATE_DOMAIN\user.name ActiveDirectory

Domain-joined, which I also suspect with #2466. This issue is also missing the `msinfo32` output which would have helped a lot with identifying this issue.

**Comment by mjgkaastrupandersen** [comment]
> Used this snippet to setup the USERNAME environment variable that finally got it to work:
> `$env:USERNAME="${env:USERDOMAIN}\${env:USERNAME}"`

Thanks. This seems to have fixed it for me.

**Comment by pbolduc** [comment]
Should be able to check `(Get-WmiObject -Class Win32_ComputerSystem).PartOfDomain` to see if the computer is domain joined and [test/fix group membership](https://github.com/code-ready/crc/blob/83eef2012d3c7af6f7d02784a100fa98d0c497e3/pkg/crc/preflight/preflight_windows.go#L184-L196) code could use `os.Getenv("USERDOMAIN")` and `os.Getenv("USERNAME")` to set member parameter like the workaround above

**Comment by anjannath** [comment]
Fixed in #2494
